tests/test_gui.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+import time
+import sys
+from GUI import *
+
+@pytest.fixture
+def gui_app():
+    # instantiate the main GUI app
+    app = mainGUI()
+    yield app
+    app.destroy()
+
+def invoke_upload_button_and_measure_time(gui_app, mocker):
+    start_time = time.time()
+
+    # mock the uploadAudio method to simulate user upload action
+    mock_upload = mocker.path('GUI.audioMenu.uploadAudio')
+
+    # invoke the upload button
+    gui_app.audioMenuList[0].uploadButton.invoke()
+
+    elapsed_time = time.time() - start_time
+    return elapsed_time
+
+def invoke_record_button_and_measure_time(gui_app, mocker):
+    start_time = time.time()
+
+    # mock recordAudio method to simulate user record action
+    mock_record = mocker.patch('GUI.audioMenu.recordAudio')
+
+    # invoke record button
+    gui_app.audioMenuList[0].recordButton.invoke()
+
+    elapsed_time = time.time() - start_time
+    return elapsed_time
+
+@pytest.mark.skipif(sys.version_info >= (3, 11, 7), reason="Only run on Python versions before 3.11.7")
```

Comment:
Let's not do the 'skip' part. Just add your test cases. Run them on a 'broken' python version and see if they fail for the right reason. Then run them on the working python version and they should pass. This way, if someone uses a broken python version, the tests will fail for that person.